### PR TITLE
undmg: 1.1.0 -> 1.1.0-unstable-2024-08-02

### PR DESCRIPTION
**Pull request by 4evy**
## Description of changes

The main reason for this PR is the addition of `ULMO/LZMA` support, which will allow more Darwin programs to be packaged with `undmg`

I tested it with Google Chrome and [Shottr](https://shottr.cc/) which use `ULMO/LZMA`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 331826` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>fmodex</li>
    <li>linuxKernel.packages.linux_4_19.prl-tools</li>
    <li>linuxKernel.packages.linux_4_19_hardened.prl-tools</li>
    <li>linuxKernel.packages.linux_5_10.prl-tools</li>
    <li>linuxKernel.packages.linux_5_10_hardened.prl-tools</li>
    <li>linuxKernel.packages.linux_5_15.prl-tools</li>
    <li>linuxKernel.packages.linux_5_15_hardened.prl-tools</li>
    <li>linuxKernel.packages.linux_5_4.prl-tools</li>
    <li>linuxKernel.packages.linux_5_4_hardened.prl-tools</li>
    <li>linuxKernel.packages.linux_6_1.prl-tools</li>
    <li>linuxKernel.packages.linux_6_10.prl-tools</li>
    <li>linuxKernel.packages.linux_6_1_hardened.prl-tools</li>
    <li>linuxKernel.packages.linux_6_6.prl-tools</li>
    <li>linuxKernel.packages.linux_hardened.prl-tools (linuxKernel.packages.linux_6_6_hardened.prl-tools)</li>
    <li>linuxKernel.packages.linux_6_8.prl-tools</li>
    <li>linuxKernel.packages.linux_6_8_hardened.prl-tools</li>
    <li>linuxKernel.packages.linux_6_9.prl-tools</li>
    <li>linuxKernel.packages.linux_6_9_hardened.prl-tools</li>
    <li>linuxKernel.packages.linux_latest_libre.prl-tools</li>
    <li>linuxKernel.packages.linux_libre.prl-tools</li>
    <li>linuxKernel.packages.linux_lqx.prl-tools</li>
    <li>linuxKernel.packages.linux_xanmod.prl-tools</li>
    <li>linuxKernel.packages.linux_xanmod_latest.prl-tools (linuxKernel.packages.linux_xanmod_stable.prl-tools)</li>
    <li>linuxKernel.packages.linux_zen.prl-tools</li>
    <li>undmg</li>
    <li>zandronum</li>
    <li>zandronum-alpha</li>
  </ul>
</details>

**Comment by DontEatOreo** [comment]
Result of `nixpkgs-review pr 331826` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>18 packages marked as broken and skipped:</summary>
  <ul>
    <li>_1password-gui</li>
    <li>_1password-gui-beta</li>
    <li>adb-sync</li>
    <li>aefs</li>
    <li>eris-go</li>
    <li>eris-go.man</li>
    <li>littlefs-fuse</li>
    <li>mpy-utils</li>
    <li>mpy-utils.dist</li>
    <li>mtpfs</li>
    <li>python311Packages.fuse</li>
    <li>python311Packages.fuse.dist</li>
    <li>python312Packages.fuse</li>
    <li>python312Packages.fuse.dist</li>
    <li>tup</li>
    <li>tup.bin</li>
    <li>tup.man</li>
    <li>unionfs-fuse</li>
  </ul>
</details>
<details>
  <summary>86 packages failed to build:</summary>
  <ul>
    <li>acquire (python312Packages.acquire)</li>
    <li>acquire.dist (python312Packages.acquire.dist)</li>
    <li>afuse</li>
    <li>backintime-common</li>
    <li>borgbackup</li>
    <li>borgbackup.dist</li>
    <li>borgbackup.doc</li>
    <li>borgbackup.man</li>
    <li>borgmatic</li>
    <li>borgmatic.dist</li>
    <li>breitbandmessung</li>
    <li>celeste</li>
    <li>deskew</li>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>dolbybcsoftwaredecode</li>
    <li>emborg (python312Packages.emborg)</li>
    <li>emborg.dist (python312Packages.emborg.dist)</li>
    <li>fpc</li>
    <li>git-annex-remote-rclone</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
    <li>libbde</li>
    <li>moosefs</li>
    <li>p4v</li>
    <li>prometheus-borgmatic-exporter</li>
    <li>prometheus-borgmatic-exporter.dist</li>
    <li>prometheus-restic-exporter</li>
    <li>python311Packages.acquire</li>
    <li>python311Packages.acquire.dist</li>
    <li>python311Packages.dissect</li>
    <li>python311Packages.dissect-target</li>
    <li>python311Packages.dissect-target.dist</li>
    <li>python311Packages.dissect.dist</li>
    <li>python311Packages.emborg</li>
    <li>python311Packages.emborg.dist</li>
    <li>python312Packages.dissect</li>
    <li>python312Packages.dissect-target</li>
    <li>python312Packages.dissect-target.dist</li>
    <li>python312Packages.dissect.dist</li>
    <li>python312Packages.llfuse</li>
    <li>python312Packages.llfuse.dist</li>
    <li>qemu-python-utils</li>
    <li>qemu-python-utils.dist</li>
    <li>rclone</li>
    <li>rclone.man</li>
    <li>reaper</li>
    <li>rectangle-pro</li>
    <li>restic</li>
    <li>resticprofile</li>
    <li>restique</li>
    <li>s3backer</li>
    <li>s3fs</li>
    <li>sbclPackages.cl-fuse</li>
    <li>sbclPackages.cl-fuse-meta-fs</li>
    <li>sbclPackages.query-fs</li>
    <li>scilab-bin</li>
    <li>securefs</li>
    <li>sensible-side-buttons</li>
    <li>sequelpro</li>
    <li>slack</li>
    <li>spacedrive</li>
    <li>spotify</li>
    <li>spotube</li>
    <li>sqlar</li>
    <li>squashfuse</li>
    <li>sshfs</li>
    <li>stats</li>
    <li>svnfs</li>
    <li>synology-drive-client</li>
    <li>telepresence2</li>
    <li>utm</li>
    <li>vlc-bin</li>
    <li>vlc-bin-universal</li>
    <li>warp-terminal</li>
    <li>wdfs</li>
    <li>wiimms-iso-tools</li>
    <li>wimlib</li>
    <li>xidel</li>
    <li>xld</li>
  </ul>
</details>
<details>
  <summary>97 packages built:</summary>
  <ul>
    <li>_9pfs</li>
    <li>acd-cli</li>
    <li>acd-cli.dist</li>
    <li>adbfs-rootless</li>
    <li>afflib</li>
    <li>android-file-transfer</li>
    <li>anki-bin</li>
    <li>apfs-fuse</li>
    <li>apparency</li>
    <li>arc-browser</li>
    <li>archivemount</li>
    <li>avfs</li>
    <li>bindfs</li>
    <li>boxfs</li>
    <li>btfs</li>
    <li>caprine-bin</li>
    <li>catcli</li>
    <li>catcli.dist</li>
    <li>chickenPackages_5.chickenEggs.fuse</li>
    <li>cryfs</li>
    <li>cryptor</li>
    <li>curlftpfs</li>
    <li>darling-dmg</li>
    <li>dbeaver-bin</li>
    <li>discord</li>
    <li>discord-canary</li>
    <li>discord-development</li>
    <li>discord-ptb</li>
    <li>dislocker</li>
    <li>dpt-rp1-py</li>
    <li>dpt-rp1-py.dist</li>
    <li>encfs</li>
    <li>exfat</li>
    <li>ext4fuse</li>
    <li>fuse (macfuse-stubs)</li>
    <li>fuse-archive</li>
    <li>fuse-ext2</li>
    <li>gitfs</li>
    <li>gitfs.dist</li>
    <li>gocryptfs</li>
    <li>grandperspective</li>
    <li>hexfiend</li>
    <li>hidden-bar</li>
    <li>hoppscotch</li>
    <li>httpdirfs</li>
    <li>httpfs2</li>
    <li>hubicfuse</li>
    <li>ifuse</li>
    <li>iina</li>
    <li>immersed-vr</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-bin)</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-bin)</li>
    <li>jmtpfs</li>
    <li>karabiner-elements</li>
    <li>karabiner-elements.driver</li>
    <li>keeweb</li>
    <li>lens</li>
    <li>libreoffice-bin</li>
    <li>lmstudio</li>
    <li>localsend</li>
    <li>mos</li>
    <li>mp3fs</li>
    <li>ntfs3g</li>
    <li>ntfs3g.dev</li>
    <li>ntfs3g.doc</li>
    <li>ntfs3g.man</li>
    <li>numi</li>
    <li>obsidian</li>
    <li>pika</li>
    <li>python311Packages.fusepy</li>
    <li>python311Packages.fusepy.dist</li>
    <li>python311Packages.htmllistparse</li>
    <li>python311Packages.htmllistparse.dist</li>
    <li>python311Packages.llfuse</li>
    <li>python311Packages.llfuse.dist</li>
    <li>python311Packages.ratarmount</li>
    <li>python311Packages.ratarmount.dist</li>
    <li>python312Packages.fusepy</li>
    <li>python312Packages.fusepy.dist</li>
    <li>python312Packages.htmllistparse</li>
    <li>python312Packages.htmllistparse.dist</li>
    <li>ratarmount (python312Packages.ratarmount)</li>
    <li>ratarmount.dist (python312Packages.ratarmount.dist)</li>
    <li>raycast</li>
    <li>realvnc-vnc-viewer</li>
    <li>rectangle</li>
    <li>roam-research</li>
    <li>romdirfs</li>
    <li>skimpdf</li>
    <li>undmg</li>
  </ul>
</details>

**Comment by JohnRTitor** [comment]
It's been three whole years since the last release, maybe ping upstream for a release? @matthewbauer

**Comment by DontEatOreo** [comment]
If upstream makes a new tag, then I will rebase, but the new revision is pretty small and only adds a new feature, so nothing major has changed, and it will still be worth upgrading (IMO) to unstable, if upstream doesn't end up making a tag. 

Will it be possible to be backported to 24.05? Since it's primarily going to serve unfree packages (e.g. Google Chrome *(potentially)* and Shottr, which is unfree too) that get backported to 24.05 with each update

File: pkgs/tools/archivers/undmg/default.nix [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
+  lzfse,
+  xz,
+  pkg-config,
+}:
 
-stdenv.mkDerivation rec {
-  version = "1.1.0";
+stdenv.mkDerivation {
   pname = "undmg";
+  version = "0-unstable-2024-08-02";
```
Comment (afh) [inline pkgs/tools/archivers/undmg/default.nix:14]: According to the [recommendations for package versions](https://github.com/NixOS/nixpkgs/tree/master/pkgs)

> If a package is a commit from a repository without a version assigned, then the version attribute should be the latest upstream version preceding that commit, followed by -unstable- and the date of the (fetched) commit. The date must be in "YYYY-MM-DD" format

this should probably be:
```suggestion
  version = "1.1.0-unstable-2024-08-02";
```

**Review by afh — CHANGES_REQUESTED** [review]
Thanks for taking this on, @DontEatOreo, please find below a suggestion that brings this package's version more in line with the current nixpkgs recommendations

**Pull request merged**